pkg/services/provisioning/dashboards/config_reader.go

```diff
@@ -51,7 +55,11 @@ func (cr *configReader) parseConfigs(file os.FileInfo) ([]*DashboardsAsConfig, e
 
 		if v0 != nil {
 			cr.log.Warn("[Deprecated] the dashboard provisioning config is outdated. please upgrade", "filename", filename)
-			return mapV0ToDashboardAsConfig(v0), nil
+			v0dashboardsAsConfig, err := mapV0ToDashboardAsConfig(v0)
```

Comment:
I think it's better to just return the results directly, since there's no benefit from the extra code:
```suggestion
			return mapV0ToDashboardAsConfig(v0)
```